File: support/domain/ddd/ValueObject.cs

namespace support.domain.ddd{
    ///
    ///<summary>Markup interface for Value Objects</summary>
    ///
    public interface ValueObject{
        ///
        ///<summary>Represents the textual information of the Value Object</summary>
        ///<returns>String with the textual representation of the Value Object</returns>
        ///
        string ToString();

        ///
        ///<summary>Represents the Value Object hashcode</summary>
        ///<returns>Integer with the Value Object hashcode</returns>
        ///
        int GetHashCode();

        ///
        ///<summary>Checks if two Value Objects are equal</summary>
        ///<param name="obj">ValueObject with the Value Object being compared to the current one</param>
        ///<returns>boolean true if both value objects are equal, false if not</returns>
        ///
        bool Equals(object obj);
    }
}

File: MYCA/myca/support/src/main/java/cdiomyc/support/domain/ddd/ValueObject.java
package cdiomyc.support.domain.ddd;

/**
 * Markup interface for Value Objects
 * @author <a href="https://github.com/freitzzz">freitzzz</a>
 */
public interface ValueObject {
    /**
     * Returns the hashcode of the value object
     * @return Integer with the hash code of the value object
     */
    @Override
    int hashCode();
    
    /**
     * Checks if a value object is equal to the current one
     * @param otherValueObject ValueObject with the comparing value object
     * @return boolean true if both value objects are equal, false if not
     */
    @Override
    boolean equals(Object otherValueObject);
    
    /**
     * Returns the textual representation of the value object
     * @return String with the textual representation of the value object
     */
    @Override
    String toString();
}
